src/unzip/seekable_http_reader.rs

```diff
@@ -393,29 +383,30 @@ impl SeekableHttpReaderEngine {
             return Ok(bytes_read_from_cache);
         }
         // - If no, check if read in progress
-        let mut read_in_progress = state.read_in_progress;
+        let mut reading_stuff = state.reader.take();
         //   Is there read in progress?
-        while read_in_progress {
+        while reading_stuff.is_none() {
             //   - If yes, release CACHE mutex, WAIT on condvar atomically
             state = self.read_completed.wait(state).unwrap();
             //     check cache again
             if let Some(bytes_read_from_cache) = state.read_from_cache(pos, buf) {
                 log::debug!("Deferred cache success");
                 return Ok(bytes_read_from_cache);
             }
-            read_in_progress = state.read_in_progress;
+            reading_stuff = state.reader.take();
         }
```

Comment:
It might be worth encapsulating this into a `State` method or other utility, in such a way that it uses RAII to put the value back. It's almost 100 lines later that this value is returned to the Option!

Reply:
I like that idea! Will do

Reply:
I had a crack at this but it turned out I didn't like the results so much. I prefer the current explicit step since then it's obvious that we put back the reading stuff _before_ releasing the mutex. I fiddled around a bit with RAII guards and with a `.with_reading_stuff(FnOnce)` type arrangement but neither was sufficiently delightful.